### PR TITLE
FIX: Prepare for PyVista 0.30.0

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
On the latest version, the API for `_add_observer()` and `use_ipyvtk` have been modified. This PR tries to catch up to those changes. Changes are also necessary in PyVistaQt.

Follows https://github.com/mne-tools/mne-python/pull/9010#issuecomment-816740584

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
Updates on [`pyvistaqt`](https://github.com/pyvista/pyvistaqt/pull/95) and [`pyvista`](https://github.com/pyvista/pyvista/pull/1257). I'm working on it.

File: tools/circleci_dependencies.sh [hunk 1]
```diff
@@ -20,7 +20,6 @@ if [[ "$CIRCLE_JOB" == "interactive_test" ]]; then
 	python -m pip install --progress-bar off --upgrade --pre --only-binary ":all:" numba llvmlite
 	wget -q https://osf.io/kej3v/download -O vtk-9.0.20201117-cp39-cp39-linux_x86_64.whl
 	python -m pip install --progress-bar off vtk-9.0.20201117-cp39-cp39-linux_x86_64.whl
-	python -m pip install --progress-bar off https://github.com/pyvista/pyvista/zipball/master
```
Comment (larsoner) [inline tools/circleci_dependencies.sh:23]: I went with a solution that should be slightly easier to eventually revert in https://github.com/mne-tools/mne-python/pull/9284/files#diff-838a9c60bcbbc8d01b84e4169048fe3fc03667b383b463c501857b168e54d082R13

Comment (GuillaumeFavelier) [reply tools/circleci_dependencies.sh:23]: Okay sure, I'll use that then

**Comment by GuillaumeFavelier** [comment]
What do you think @larsoner ?

**Comment by larsoner** [comment]
Thanks @GuillaumeFavelier !

**Pull request merged**